itests/src/test/java/fi/metatavu/edelphi/test/mock/QueryMocker.java

```diff
@@ -0,0 +1,272 @@
+package fi.metatavu.edelphi.test.mock;
+
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fi.metatavu.edelphi.test.mock.QueryPageMock.SettingType;
+
+public class QueryMocker extends AbstractResourceMocker {
+
+  private static Logger logger = Logger.getLogger(QueryMocker.class.getName());
+
+  private List<Long> queryIds = new ArrayList<>();
+  private List<Long> sectionIds = new ArrayList<>();
+  private List<PageMeta> pages = new ArrayList<>();
+  private List<Long> pageSettingIds = new ArrayList<>();
+
+  public QueryMocker mock() {
+    return this;
+  }
+
+  public QueryMocker cleanup() {
+    for (Long pageSettingId : pageSettingIds) {
+      deleteQueryPageSetting(pageSettingId);
+    }
+
+    for (PageMeta page : pages) {
+      deleteQueryPage(page.getId());
+    }
+
+    for (Long sectionId : sectionIds) {
+      deleteQuerySection(sectionId);
+    }
+
+    for (Long queryId : queryIds) {
+      deleteQuery(queryId);
+    }
+
+    return this;
+  }
+
+  public long addQuery(String name, String urlName, Long parentFolderId, boolean visible, String description,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)